File: android/app/src/main/java/com/purchasely/shaker/domain/model/Cocktail.kt
package com.purchasely.shaker.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class CocktailsData(
    val cocktails: List<Cocktail>
)

@Serializable
data class Cocktail(
    val id: String,
    val name: String,
    val image: String,
    val description: String,
    val category: String,
    val spirit: String,
    val difficulty: String,
    val tags: List<String>,
    val ingredients: List<Ingredient>,
    val instructions: List<String>
)

@Serializable
data class Ingredient(
    val name: String,
    val amount: String
)

File: ios/Shaker/Model/Cocktail.swift
import Foundation

struct CocktailsData: Codable {
    let cocktails: [Cocktail]
}

struct Cocktail: Codable, Identifiable {
    let id: String
    let name: String
    let image: String
    let description: String
    let category: String
    let spirit: String
    let difficulty: String
    let tags: [String]
    let ingredients: [Ingredient]
    let instructions: [String]
}

struct Ingredient: Codable, Identifiable {
    let name: String
    let amount: String

    var id: String { name }
}
